nifi-docs/src/main/asciidoc/developer-guide.adoc

```diff
@@ -2296,6 +2296,32 @@ threads that should be used to run the Processor can
 be set via the `setThreadCount(int)` method.
 
 
+=== Experimental NiFi Flow test harness
+
+NiFi now has an experimental feature for full end-to-end testing of flows. This allows us
+to take a NiFi flow, install it to a test NiFi instance, run it and make Java unit test
+like asserts regarding its behaviour.
+
+The class `org.apache.nifi.test.TestNiFiInstance` is a thin wrapper that allows us
+to manipulate a NiFi installation and deploy a flow with some adjustments
+to its configuration, including changing processor properties and replacing processor
+classes with mocks.
+
+In order to add the necessary classes to your project,
```

Comment:
Just to clarify, you mean external projects that use NiFi. To use this with NiFi's code base, a developer would have to add something to the end of the build system that would run the tests after the assembly is built based on what I've seen reviewing this. Is that right or am I missing something?

Just want to make sure in case we have to answer questions from people come in and try to use this to test their contributions.

Reply:
Yes, absolutely. After reading the paragraph once again, I'm wondering if it would make sense to change `us` to `you` (referring to the user), but it is you call, really. 
If you think it makes sense, can you do it while you merge/squash the commit?